lib/discourse_plugin_registry.rb

```diff
@@ -50,6 +50,16 @@ def self.define_filtered_register(register_name)
     define_singleton_method("register_#{register_name.to_s.singularize}") do |value, plugin|
       public_send(:"_raw_#{register_name}") << { plugin: plugin, value: value }
     end
+
+    # Special case for reviewable_types, as we need to be able to lookup the plugins
+    # that register each type.
+    if register_name.to_s == "reviewable_types"
```

Comment:
I ran into plugin errors if allowed the `_lookup` helper for all register names (for example, the discourse-ai spam scanner would cause errors when publishing a new post). Since we only need this helper for `reviewable_types`, the most straightforward fix was to restrict it just to that type.

That said, since this is still touching some fairly core APIs, I'd appreciate some extra testing to make sure I haven't missed anything.

Reply:
Instead of adding a specific conditional here, perhaps the `define_filtered_register` can accept a block or keyword arg to define this lookup method so as to keep the method generic.